### PR TITLE
samples: bluetooth: fast_pair: input_device: Add init nRF54L support 

### PR DESCRIPTION
**Pull request by mkapala-nordic**
Depends on: 
* https://github.com/nrfconnect/sdk-nrf/pull/14285 -> buttons workaround
* https://github.com/nrfconnect/sdk-nrf/pull/14368 -> partition manager for nRF54L fix
* https://github.com/nrfconnect/sdk-nrf/pull/14409 -> partition manager for nRF54L support in sysbuild

Jira: NCSDK-26123

### CODE REVIEW

File: subsys/bluetooth/services/fast_pair/CMakeLists.txt [hunk 1]
```diff
@@ -51,41 +51,69 @@ if(NOT SYSBUILD)
...
+                          -m ${FP_MODEL_ID} -k ${FP_ANTI_SPOOFING_KEY}
+      COMMAND_ECHO STDOUT
+      )
+
+    set_property(
+      GLOBAL APPEND PROPERTY
+      HEX_FILES_TO_MERGE
+      ${FP_PROVISIONING_DATA_HEX}
+      ${CMAKE_BINARY_DIR}/zephyr/zephyr.hex
+      )
```
Comment (nordicjm) [inline subsys/bluetooth/services/fast_pair/CMakeLists.txt:109]: I don't really have time at present to go through this in detail or check it but at first glance this does not seem right, it assumes that the output is names zephyr.hex, which is not the case. The output filename is set with a Kconfig, and there's also the possibility of using MCUboot signing which generates a signed image

Comment (mkapala-nordic) [reply subsys/bluetooth/services/fast_pair/CMakeLists.txt]: > The output filename is set with a Kconfig

Alright, I was not aware of the `CONFIG_KERNEL_BIN_NAME` Kconfig option. Thanks, fixed.

> there's also the possibility of using MCUboot signing which generates a signed image

I think that MCUboot forces to use Partition Manager so should not be a problem here as this path is only valid for the `CONFIG_PARTITION_MANAGER_ENABLED=n`

**Review by SeppoTakalo — APPROVED** [review]

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/services/fast_pair/CMakeLists.txt [hunk 2]
```diff
@@ -51,41 +51,69 @@ if(NOT SYSBUILD)
...
+      ${CMAKE_BINARY_DIR}/zephyr/${CONFIG_KERNEL_BIN_NAME}.hex
+      )
+
+    set_property(
+      TARGET
+      runners_yaml_props_target
+      PROPERTY
+      hex_file
+      ${CMAKE_BINARY_DIR}/zephyr/merged.hex
+      )
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/bluetooth/services/fast_pair/CMakeLists.txt:117]: E.g. try to add Zephyr hex file again to replicate a build error

```
    set_property(                                                                                    
      GLOBAL APPEND PROPERTY                                                                         
      HEX_FILES_TO_MERGE                                                                             
      ${CMAKE_BINARY_DIR}/zephyr/${CONFIG_KERNEL_BIN_NAME}.hex                                       
      )
```

**Review by alstrzebonski — COMMENTED** [review]
No comments from me apart from Marek's comments. I will rereview once they will be addressed.

**Review by nordicjm — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]

**Pull request merged**